lib/utils/Utils.js

```diff
@@ -154,10 +157,11 @@ export default class Utils {
       throw new Error('Expected an array of exactly 1 array element as the JMAP response for a "' + request + '" request.');
     }
 
-    let expectedResponse = Utils._expectedResponseFor(request);
+    let response = data[0][0],
+        expectedResponse = Utils._expectedResponseFor(request);
 
-    if (expectedResponse && data[0][0] !== expectedResponse) {
-      throw new Error('Expected "' + expectedResponse + '" as the JMAP response for a "' + request + '" request, but got "' + data[0][0] + '".');
+    if (response !== ERROR && (expectedResponse && response !== expectedResponse)) {
```

Comment:
what is the diff between
`(response !== ERROR && (expectedResponse && response !== expectedResponse`
and
`(response !== ERROR && response !== expectedResponse`

why do I need to verify that `expectedResponse!== undefined`?